sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_keep_test.c

```diff
@@ -0,0 +1,114 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/drivers/retention_sram.h"
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+#include "sw/device/silicon_creator/lib/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kPattern = 0xab,
+};
+
+/**
+ * Check that the values in the retention SRAM have not changed
+ * from kPattern. The 0th word of the SRAM is exempt as it is
+ * used to store the reset reason.
+ */
+bool check_ram_unchanged(retention_sram_t *ret) {
+  LOG_INFO("Checking that retention SRAM values are unchanged");
+  static uint32_t raw[sizeof(retention_sram_t) / sizeof(uint32_t)];
+  uint32_t pattern32;
+  memset(&pattern32, kPattern, sizeof(pattern32));
+
+  // Ensure that all written sections were saved
+  memcpy(raw, ret, sizeof(retention_sram_t));
+
+  bool unchanged = true;
+  // Skip checking index 0 since that is used to store the
+  // reset reason and will be changed.
+  for (size_t i = 1; i < ARRAYSIZE(raw); ++i) {
+    if (raw[i] != pattern32) {
+      LOG_ERROR("Retention SRAM changed at offset %u (%x --> %x).", i,
+                pattern32, raw[i]);
+      unchanged = false;
+    }
+  }
+  return unchanged;
+}
+
+rom_error_t retention_ram_keep_test(void) {
+  // Variables of type `retention_sram_t` are static to reduce stack usage.
+  static retention_sram_t *ret;
+  ret = retention_sram_get();
```

Comment:
nit: this doesn't need to be `static` since this is just a pointer.
```suggestion
  retention_sram_t *ret = retention_sram_get();
```